src/Npgsql/PgPassEntry.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Npgsql
+{
+    /// <summary>
+    /// Represents a hostname, port, database, username, and password combination that has been retrieved from a .pgpass file
+    /// </summary>
+    class PgPassEntry
+    {
+        #region Fields and Properties
+
+        /// <summary>
+        /// Hostname parsed from the .pgpass file
+        /// </summary>
+        internal string Hostname { get; }
+        /// <summary>
+        /// Port parsed from the .pgpass file
+        /// </summary>
+        internal string Port { get; }
```

Comment:
It would be cleaner to have this as an int rather than a string, after all this class is aware of the actual structure of the pgpass file and therefore also of the typing.